features/integrate_build_system/waf-generator/conanfile.py

```diff
@@ -0,0 +1,39 @@
+from conans.model import Generator
+from conans import ConanFile
+
+
+class Waf(Generator):
+    def _remove_lib_extension(self, libs):
+        return [lib[0:-4] if lib.endswith(".lib") else lib for lib in libs]
+
+    @property
+    def filename(self):
+        return "waf_conan_libs_info.py"
+
+    @property
+    def content(self):
+        sections = []
+        sections.append("def configure(ctx):")
+        self.deps_build_info.libs = self._remove_lib_extension(
+            self.deps_build_info.libs)
+        conan_libs = []
+        for dep_name, info in self.deps_build_info.dependencies:
+            if dep_name not in self.conanfile.build_requires:
+                info.libs = self._remove_lib_extension(info.libs)
```

Comment:
I will try not to modify the ``info`` object, as that is normally used only to extract the information. Better use a local variable or do it all at once in the ``sections.append("   ctx.env.CONAN_LIBS ...... )``